plugins/webapi/ledgerstate/output.go

```diff
@@ -0,0 +1,168 @@
+package ledgerstate
+
+import (
+	"net/http"
+
+	"github.com/iotaledger/goshimmer/packages/ledgerstate"
+	"github.com/iotaledger/goshimmer/plugins/messagelayer"
+	"github.com/labstack/echo"
+	"golang.org/x/xerrors"
+)
+
+// region API endpoints ////////////////////////////////////////////////////////////////////////////////////////////////
+
+// GetOutputEndPoint is the handler for the /ledgerstate/outputs/:outputID endpoint.
+func GetOutputEndPoint(c echo.Context) error {
+	outputID, err := ledgerstate.OutputIDFromBase58(c.Param("outputID"))
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(err))
+	}
+
+	cachedOutput := messagelayer.Tangle().LedgerState.Output(outputID)
+	defer cachedOutput.Release()
+	// output object is nil if was not found
+	output := cachedOutput.Unwrap()
+	if output == nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(xerrors.Errorf("output not present in node")))
+	}
+
+	return c.JSON(http.StatusOK, NewOutput(output))
+}
+
+// GetOutputConsumersEndPoint is the handler for the /ledgerstate/outputs/:outputID/consumers endpoint.
+func GetOutputConsumersEndPoint(c echo.Context) (err error) {
+	outputID, err := ledgerstate.OutputIDFromBase58(c.Param("outputID"))
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(err))
+	}
+
+	cachedConsumers := messagelayer.Tangle().LedgerState.Consumers(outputID)
+	defer cachedConsumers.Release()
+
+	consumers := cachedConsumers.Unwrap()
+	// check all returned and unwrapped objects
+	for _, consumer := range consumers {
+		if consumer == nil {
+			return c.JSON(http.StatusBadRequest, NewErrorResponse(xerrors.Errorf("failed to load consumers")))
+		}
+	}
+
+	return c.JSON(http.StatusOK, NewConsumers(consumers))
+}
+
+// GetOutputMetadataEndPoint is the handler for the /ledgerstate/outputs/:outputID/metadata endpoint.
+func GetOutputMetadataEndPoint(c echo.Context) (err error) {
+	outputID, err := ledgerstate.OutputIDFromBase58(c.Param("outputID"))
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(err))
+	}
+
+	cachedOutputMetadata := messagelayer.Tangle().LedgerState.OutputMetadata(outputID)
+	defer cachedOutputMetadata.Release()
+	// outputMetadata object is nil if was not found
+	outputMetadata := cachedOutputMetadata.Unwrap()
+	if outputMetadata == nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(xerrors.Errorf("output metadata not present in node")))
+	}
+
+	return c.JSON(http.StatusOK, NewOutputMetadata(outputMetadata))
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region Output ///////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Output represents a JSON model of a ledgerstate.Output.
+type Output struct {
+	ID       string           `json:"id"`
```

Comment:
I think it would be nice to also explicitly show the TransactionID and the OutputIndex here by using the `OutputID` model from `NewOutputID` instead of just a string.

In code you can derive that from the OutputID so we don't have that as properties in the model but since this logic is not available to a potential API client, it would be nice to just spell these two values out in the JSON model.

Reply:
done!